opm/autodiff/BlackoilModelBase_impl.hpp

```diff
@@ -180,15 +180,23 @@ namespace detail {
         , terminal_output_ (terminal_output)
     {
 #if HAVE_MPI
-        if ( terminal_output_ ) {
-            if ( linsolver_.parallelInformation().type() == typeid(ParallelISTLInformation) )
-            {
-                const ParallelISTLInformation& info =
-                    boost::any_cast<const ParallelISTLInformation&>(linsolver_.parallelInformation());
+        if ( linsolver_.parallelInformation().type() == typeid(ParallelISTLInformation) )
+        {
+            const ParallelISTLInformation& info =
+                boost::any_cast<const ParallelISTLInformation&>(linsolver_.parallelInformation());
+            if ( terminal_output_ ) {
                 // Only rank 0 does print to std::cout if terminal_output is enabled
                 terminal_output_ = (info.communicator().rank()==0);
             }
+            int global_number_of_wells = wells_->number_of_wells;
```

Comment:
At this point, it is the local, not global number of wells, so this variable should be renamed, and a separate (global) used for the sum.